packages/remix-react/components.tsx

```diff
@@ -83,55 +82,6 @@ function useRemixContext(): RemixContextObject {
   return context;
 }
 
-////////////////////////////////////////////////////////////////////////////////
-// RemixRoute
-
-export function RemixRoute({ id }: { id: string }) {
```

Comment:
`RemixRoute` and `RemixRouteError` were sort of poor-mans implementations of `Component`/`ErrorBoundary` lazy loading.  We could assign them to a route definition up front (without a route module available) knowing that they'd pull from the lazily loaded `routeModulesCache` at render-time.  Now we can just assign `Component`/`ErrorBoundary` via `lazy()`.

The only downside is we lose these error messages - so if we think they are worth keeping we can stick with this approach.